src/components/Insertion/InsertionDisplay/index.tsx

```diff
@@ -16,24 +15,28 @@ const InsertionDisplay: React.FC = () => {
         where('userId', '==', user?.uid),
       );
       const querySnapshot = await getDocs(q);
-
-      const data = querySnapshot.docs.map(doc => {
+      const insertions = querySnapshot.docs.map(doc => {
         const docData = doc.data();
         return {
           id: doc.id,
           ...docData,
         } as InsertionData;
       });
 
-      setInsertions(data);
+      return insertions;
     } catch (error) {
       console.error('Error fetching documents:', error);
     }
   };
 
-  useEffect(() => {
-    getData();
-  }, []);
+  const { data: insertions, isLoading } = useQuery({
+    queryFn: getData,
+    queryKey: ['insertions'],
```

Comment:
questa key non e' abbastanza specifica

la chiave delle query serve per identificarle univocamente, nel senso che a parita di chiave, tanstack query riusa i dati che ha

concretamente vuol dire che __tu stai fetchando i dati dell'utente, ma la chiave non e' specifica all'utente__

quindi siccome stai usando solo `["insertions"]`, se vai nel profilo di un'altra persona, o nella home page, o ovunque, la key non cambia e tanstack query riusera questi dati che gia ha (presi dalla cache)

ora, e' difficile uscire una metodologia di naming per le chiavi in poche righe, pero per ora ti importa solo che devono essere specifiche alle dipendenze della chiamata

per questa chiamata una roba tipo `["insertions", "byUser", user.uid]` potrebbe andare, mentre nella ricerca sarebbe tipo `["insertions", "bySearchQuery", searchQuery]`

Reply:
Perfetto. Allo stesso modo quindi dovrei cambiare questa?

 `const { data: insertions, isLoading } = useQuery({
    queryFn: getInsertions,
    queryKey: ['insertions', 'byQuery', query],
  });`

Reply:
si